src/halfpipe/tui/base.py

```diff
@@ -0,0 +1,254 @@
+                left_button_text="YES",
+                right_button_text="NO",
+                left_button_variant="error",
+                right_button_variant="success",
+                title="Quit?",
+                id="quit_modal",
+                classes="confirm_warning",
+            ),
+            quit,
+        )
+
+    def render(self) -> RenderResult:
+        """Render the header icon.
+
+        Returns:
+            The rendered icon.
+        """
+        return self.icon
+
+
+class HeaderHelpIcon(Widget):
+    """Display an 'icon' on the left of the header."""
+
+    DEFAULT_CSS = """
+    HeaderHelpIcon {
+        dock: right;
+        padding: 0 1;
+        width: 8;
+        content-align: left middle;
+        offset-x: -3;
+    }
+
+    HeaderHelpIcon:hover {
+        background: $foreground 10%;
+    }
+    """
+    """The character to use as the icon within the header."""
+    icon = Reactive("❓")
+    help_string = "Here should be some general help :) Or maybe link to manual?"
+
+    async def on_click(self, event: Click) -> None:
+        """Launch the command palette when icon is clicked."""
+        event.stop()
+        await self.app.push_screen(
+            Confirm(
+                self.help_string,
+                left_button_text=False,
+                right_button_text="OK",
+                right_button_variant="default",
+                title="Help",
+                id="help_modal",
+                #   classes="confirm_warning",
+            )
+        )
+
+    def render(self) -> RenderResult:
+        """Render the header icon.
+
+        Returns:
+            The rendered icon.
+        """
+        return self.icon
+
+
+class MyHeader(Header):
+    def compose(self):
+        yield HeaderTitle()
+        yield HeaderHelpIcon()
+        yield HeaderCloseIcon()
+
+
+class RichImage:
+    """Convert the image to a Rich image."""
+
+    # def __rich_console__(self, console: Console, options) -> RenderResult:
+    #     with Image.open(os.path.join(BASE_DIR, "images/halfpipe_logo_v2.png")) as image:
+    #         pixels = Pixels.from_image(image, resize=(110, 92))  # 105, 92
+    #     return console.render(pixels)
+
+
+class ImageContainer(Container):
+    """Create a container for the image."""
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+    def render(self):
+        return RichImage()
+
+    def key_escape(self):
+        self.dismiss(False)
+
+
+class Welcome(ModalScreen):
+    """Intro screen with an intro image."""
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+    def compose(self) -> ComposeResult:
+        yield ImageContainer(id="welcome_image")
+
+    def key_escape(self):
+        self.dismiss(True)
+
+    def on_click(self, event: events.Click) -> None:
+        self.dismiss(True)
+
+
+class MainApp(App):
+    CSS_PATH = [
+        os.path.join(BASE_DIR, "tcss/base.tcss"),
```

Comment:
Please use `pathlib` for joining paths, as it is the current best practice. It should only be a simple search and replace
```suggestion
        Path(BASE_DIR) / "tcss/base.tcss",
```